lib/fuse.c

```diff
@@ -4702,6 +4705,12 @@ struct fuse *fuse_new_31(struct fuse_args *args,
 		}
 	}
 
+
+	if (f->conf.max_idle_threads == -1) {
+		/* If the number of idle threads was not set, default to 10. */
+		f->conf.max_idle_threads = 10;
+	}
+
```

Comment:
Setting the value to 0 should probably not be allowed. At least warned about. I'm not even sure this dynamic behavior should be done at all. Not without smarter logic around culling unneeded threads. Spawning a thread on demand is costly. This isn't a Erlang or some other light weight process. I've seen my system thrash due to in part to libfuse rapidly spawning new threads and then killing them as soon as they finished with their task. In my branch of libfuse I've completely removed that logic and made a fixed pool that is user configurable and defaults to the number of cores. By doing so it removes all those mutexes and simplifies the code quite a bit.

In my tests I also found that N - 1 threads led to the best performance in general. Both in situations where there was one app hammering the FS and where there were several. (N = number of physical cores, not logical).

I'm all for this change to enable setting the number of threads but it should be well documented how the threads come into and go out of existence.